flagsmith-core.ts

```diff
@@ -681,12 +661,86 @@ const Flagsmith = class {
     };
 
+    trackEvent = (event: string) => {
+        if (!this.splitTestingAnalytics) {
+            const error = new Error("This feature is only enabled for self-hosted customers using split testing.");
+            console.error(error.message);
+            return Promise.reject(error);
+        } else if (!this.identity) {
+            this.events.push(event);
+            this.log("Waiting for user to be identified before tracking event", event);
+            return Promise.resolve();
+        } else {
+            return this.analyticsFlags().then(() => {
+                return this.getJSON(this.api + 'split-testing/conversion-events/', "POST", JSON.stringify({
+                    'identity_identifier': this.identity,
+                    'type': event
+                }));
+            });
+        }
+    };
+
+    private _loadedState(error: any = null, source: FlagSource, isFetching = false) {
+        return {
+            error,
+            isFetching,
+            isLoading: false,
+            source
+        }
+    }
+
+    private analyticsFlags = () => {
+        const { api } = this;
+
+        if (!this.evaluationEvent|| !this.evaluationEvent[this.environmentID] || !api) {
+            return Promise.resolve()
+        }
+
+        if (this.evaluationEvent && Object.getOwnPropertyNames(this.evaluationEvent).length !== 0 && Object.getOwnPropertyNames(this.evaluationEvent[this.environmentID]).length !== 0) {
+            return this.getJSON(apiVersion(`${api}`, this.splitTestingAnalytics?2:1) + 'analytics/flags/',
+                'POST',
+                JSON.stringify(this.parseV2Analytics(this.evaluationEvent[this.environmentID])))
+                .then((res) => {
+                    const state = this.getState();
+                    if (!this.evaluationEvent) {
+                        this.evaluationEvent = {}
+                    }
+                    this.evaluationEvent[this.environmentID] = {}
+                    this.setState({
+                        ...state,
+                        evaluationEvent: this.evaluationEvent,
+                    });
+                    this.updateEventStorage();
+                }).catch((err) => {
+                    this.log("Exception fetching evaluationEvent", err);
+                });
+        }
+        return Promise.resolve()
+    };
+
     private log(...args: (unknown)[]) {
         if (this.enableLogs) {
             console.log.apply(this, ['FLAGSMITH:', new Date().valueOf() - (this.timer || 0), 'ms', ...args]);
         }
     }
 
+    private parseV2Analytics = (evaluations: Record<string, number>|null)=> {
+        if(!this.splitTestingAnalytics) {
+            return evaluations || {}
+        }
+        if(!evaluations) return {evaluations: []}
+        return {
+            evaluations:  Object.keys(evaluations).map((feature_name)=>(
+                {
+                    feature_name,
+                    "identity_identifier": this.identity||null,
```

Comment:
Unless I'm missing something, the current `this.identity` here could be different than the one when the event happened, for example if the user logged out/in before analytics were flushed.

The same applies to `hasFeature` below - haven't reviewed this in much detail so might be completely off

Reply:
Hmmm yeah I suppose they should be more of a snapshot, I'll have a think about this / look if there are any nasty race conditions